src/coprocessor/dag/dag.rs

```diff
@@ -113,6 +112,89 @@ impl DAGContext {
         }
     }
 
+    pub fn handle_streaming_request(&mut self, batch_row_limit: usize) -> Result<Option<Response>> {
+        let mut record_cnt = 0;
```

Comment:
Could we return directly if `self.finished=true` here?

Reply:
In my design we will never call `handle_streaming_request` if `finish == true`. Maybe move the status tracker out from `DAGContext` is better.